base/system/services/database.c

```diff
@@ -1364,23 +1468,24 @@ ScmControlService(HANDLE hControlPipe,
                         &Overlapped);
     if (bResult == FALSE)
     {
-        DPRINT("WriteFile() returned FALSE\n");
+        DPRINT("(%S, %d) WriteFile() returned FALSE\n", lpService->lpServiceName, dwControl);
 
         dwError = GetLastError();
         if (dwError == ERROR_IO_PENDING)
         {
-            DPRINT("dwError: ERROR_IO_PENDING\n");
+            DPRINT("(%S, %d) dwError: ERROR_IO_PENDING\n", lpService->lpServiceName, dwControl);
 
             dwError = WaitForSingleObject(hControlPipe,
                                           PipeTimeout);
-            DPRINT("WaitForSingleObject() returned %lu\n", dwError);
+            DPRINT("(%S, %d) WaitForSingleObject() returned %lu\n", lpService->lpServiceName, dwControl, dwError);
 
             if (dwError == WAIT_TIMEOUT)
             {
+                DPRINT1("(%S, %d) WaitForSingleObject() timed out!\n", lpService->lpServiceName, dwControl, dwError);
                 bResult = CancelIo(hControlPipe);
                 if (bResult == FALSE)
                 {
-                    DPRINT1("CancelIo() failed (Error: %lu)\n", GetLastError());
+                    DPRINT("(%S, %d) CancelIo() failed (Error: %lu)\n", lpService->lpServiceName, dwControl, GetLastError());
```

Comment:
Just a question out of curiosity: Whey the DPRINT1->DPRINT in this error-scenario? Did it spam?

Reply:
It did not spam. In general, after these changes, the code seems to work well and there is not any need for printing detailed failure messages, so demoting the DPRINT1's to DPRINT's seemed reasonable.